### DISCUSSION TITLE
Left Side of Line Cutting Off

### DISCUSSION THREAD

**Discussion by sean-brown-dev** [patrykandpatrick/vico#489] [Questions] [answered]
I'm not sure if this is an issue yet or not so I'm just going to ask in the discussion section. I have the same chart using the same data on two different screens in my Jetpack Compose app. On one screen the line on the left side of the chart always cuts ends at the last point for some reason. The one on the other screen correctly draws the line all the way to the edge when there's another point to the left. I have posted screenshots of both charts. Has anyone seen this before and do you know how to fix it?

Edit: I noticed on the right side that it is not cutting off the line, but the line shifts. It seems like the data points are not being included anymore when they go out of view. On the left side it cuts off the line and on the right the line just goes off to some different location instead of towards the next point. 

![Screenshot_20231211-192836.png](https://github.com/patrykandpatrick/vico/assets/60934413/25aef87f-1cf2-4030-a2d5-4e67f1b7e3c1)

![Screenshot_20231211-192812.png](https://github.com/patrykandpatrick/vico/assets/60934413/6794d538-c233-4fbf-b70c-f4addb7e4f60)

**Comment by patrickmichalik** [comment]
Hello! I posted a comment here a few hours ago, saying that we’re working on a fix. I then deleted it. Having taken a closer look, I’m under the impression that the bug we’re working on is either only partially related or not related at all. Something else seems to be happening here.

Would you be able to share screen recordings of both charts? In both cases, please start the recording with the chart scrolled all the way to the left and then scroll the chart all the way to the right. This would give us the best idea of what’s going on. Also, could you please let us know what version of Vico you’re using?

**Reply by sean-brown-dev** [reply]
Thanks for the quick reply! I've attached videos of both charts and if you need I can give any code snippets as well. I'm using the same composable for both and building the chart model using the same code as well, so I'm pretty stumped.

I am using v1.13.0

I'm hitting an upload size limit for the videos, so below are links.

Broken:
https://drive.proton.me/urls/TSSFQB7QA8#feWS-jAWDFxE

Working:
https://drive.proton.me/urls/VA8K7X2ATC#aI8iPKat9M9t

**Reply by patrickmichalik** [reply]
Thanks, @sean-brown-dev! That’s certainly rather odd. If you could share the chart code, we’d appreciate it.

**Reply by sean-brown-dev** [reply]
Sure thing! The ChartModel.kt file is where I build the models. `getVolumeChartModel()` is where I build the one specifically for this chart, but the other line charts in there have the same scrolling cutoff issue as well. The other two files are the composable where I create the charts in the UI by passing in the chart models.

[Charts.zip](https://github.com/patrykandpatrick/vico/files/13651463/Charts.zip)

**Reply by patrickmichalik** [reply, marked as answer]
Thanks! Could it be the case that the entries aren’t sorted? If so, please try adding `sortBy { it.date.toEpochDay() }` to the `volumesByLocalDate` chain, before `associateBy`.

For any two entries in a single series, the entry whose index in the series is greater should have a greater _x_ value. Vico 2.0.0 Alpha 4, which should be out shortly, will take care of this automatically. It’s relatively unlikely that this is the issue here—I’d expect to see more visual glitches if it were—but it might be worth checking this before we move on to more complex troubleshooting.

**Reply by sean-brown-dev** [reply]
That was it! I sort all of the logs before passing it to those methods, but I am sorting it by descending rather than ascending (the same list of logs is used elsewhere where it needed to be descending). Thanks a bunch I never would have guessed that one!

**Reply by patrickmichalik** [reply]
Ah, that explains why there were no other glitches—each entry had the right neighbors. I’m glad we’ve figured it out. You’re welcome! By the way, [Vico 2.0.0 Alpha 4](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.4), which I mentioned above, is now available.
